### ISSUE TITLE
compilation error in StepRegistrationTest

### ISSUE THREAD

**Issue by aswinthomas** [cucumber/cucumber-cpp#78]
My build failed at StepRegistrationTest. 
```
[ 90%] Building CXX object tests/CMakeFiles/StepRegistrationTest.dir/integration/StepRegistrationTest.cpp.o
In file included from /home/aswin/apps/cucumber-cpp/tests/integration/StepRegistrationTest.cpp:1:0:
/usr/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult::AssertionResult(const T&, typename testing::internal::EnableIf<(! testing::internal::ImplicitlyConvertible<T, testing::AssertionResult>::value)>::type*) [with T = cucumber::internal::MatchResult; typename testing::internal::EnableIf<(! testing::internal::ImplicitlyConvertible<T, testing::AssertionResult>::value)>::type = void]’:
/home/aswin/apps/cucumber-cpp/tests/integration/StepRegistrationTest.cpp:21:5:   required from here
/usr/include/gtest/gtest.h:277:25: error: passing ‘const cucumber::internal::MatchResult’ as ‘this’ argument of ‘cucumber::internal::MatchResult::operator bool()’ discards qualifiers [-fpermissive]
       : success_(success) {}
                         ^
make[2]: *** [tests/CMakeFiles/StepRegistrationTest.dir/integration/StepRegistrationTest.cpp.o] Error 1
make[1]: *** [tests/CMakeFiles/StepRegistrationTest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Same happens for TaggedHookRegistrationTest and WireServerTest.

I also had to add pthread to line 39 tests/CMakeLists.txt due to a linker issue

**Comment by paoloambrosio** [comment]
What OS, compiler and libraries are you using?

**Comment by aswinthomas** [comment]
Sorry I dint mention that. I am using the following:
- Ubuntu 14.04 (I understand this is not supported yet)
- g++ 4.8.2
- latest from source gtest, gmock and cppspec

For now I have commented out compilation of the specifc tests.

**Comment by konserw** [comment]
Probably fixed in #82 #92 #94 
In my opinion can be closed

**Comment by paoloambrosio** [comment]
Now it compiles on Travis. Please reopen if not fixed.
